src/Microsoft.DotNet.Build.Tasks.Packaging/src/GetLastStablePackage.cs

```diff
@@ -0,0 +1,104 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using NuGet.Versioning;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Build.Tasks.Packaging
+{
+    public class GetLastStablePackage : PackagingTask
+    {
+        /// <summary>
+        /// List of packages to look up.  Stable version less than this version will be returned.
+        ///   Identity: Package ID
+        ///   Version: Package version.
+        /// </summary>
+        [Required]
+        public ITaskItem[] LatestPackages { get; set; }
+        
+        /// <summary>
+        /// List of previously shipped packages.
+        ///   Identity: Package ID
+        ///   Version: Package version.
+        /// </summary>
+        [Required]
+        public ITaskItem[] StablePackages { get; set; }
+
+        /// <summary>
+        /// Latest version from StablePackages for all packages in LatestPackages.
+        /// If a version isn't found for an item in LatestPackage that will not be included in this set.
+        /// </summary>
+        [Output]
+        public ITaskItem[] LastStablePackages { get; set; }
+
+        public override bool Execute()
+        {
+            if (LatestPackages == null || LatestPackages.Length == 0)
+            {
+                return true;
+            }
+
+            Dictionary<string, Version> latestPackages = new Dictionary<string, Version>();
+            Dictionary<string, Version> lastStablePackages = new Dictionary<string, Version>();
+            
+            foreach(var latestPackage in LatestPackages)
+            {
+                var packageId = latestPackage.ItemSpec;
+
+                var versionString = latestPackage.GetMetadata("Version");
+                NuGetVersion nuGetVersion = null;
+                if (versionString == null || !NuGetVersion.TryParse(versionString, out nuGetVersion))
+                {
+                    Log.LogMessage($"Could not parse version {versionString} for LatestPackage {packageId}, will use latest stable.");
+                }
+
+                latestPackages[packageId] = nuGetVersion?.Version;
+            }
+
+            foreach(var stablePackage in StablePackages)
+            {
+                var packageId = stablePackage.ItemSpec;
+
+                Version latestVersion;
+                if (!latestPackages.TryGetValue(packageId, out latestVersion))
+                {
+                    continue;
```

Comment:
>continue; [](start = 20, length = 9)

I'm not sure I understand when this can happen? In theory if you have a stable package there should be an entry in latestPackages already for it, or not?